src/routes/collectibles/handlers.rs

```diff
@@ -30,3 +36,76 @@ pub async fn collectibles(
             .await?,
     ))
 }
+
+/// Returns paginated collectibles.
+pub async fn collectibles_paginated(
+    context: &RequestContext,
+    chain_id: &str,
+    safe_address: &str,
+    cursor: &Option<String>,
+    trusted: Option<bool>,
+    exclude_spam: Option<bool>,
+) -> ApiResult<Json<Page<ServiceCollectible>>> {
+    let info_provider = DefaultInfoProvider::new(chain_id, &context);
+    let page_metadata = cursor
+        .as_ref()
+        .map(|cursor| PageMetadata::from_cursor(cursor));
+
+    let url = core_uri!(
+        info_provider,
+        "/v2/safes/{}/collectibles/?{}&trusted={}&exclude_spam={}", /* paginated service in core is behind v2 api. */
+        safe_address,
+        page_metadata
+            .as_ref()
+            .unwrap_or(&PageMetadata::default())
+            .to_url_string(),
```

Comment:
nitpick: I'd move this up where you map from the `cursor` (lines 50 to 52) since we don't need the `page_metadata` for anything else

Reply:
Clearest